js/coinsbit.js

```diff
@@ -0,0 +1,799 @@
+            }
+            data = this[method] (this.deepExtend (request, params));
+            // [
+            //     {
+            //         tid: 99601,
+            //         type: "buy",
+            //         date: 1640108002,
+            //         amount: "323144.05252658",
+            //         price: "0.008",
+            //         total: "2585.15242021"
+            //     },
+            //     {
+            //         tid: 99544,
+            //         type: "buy",
+            //         date: 1640107990,
+            //         amount: "967.91335",
+            //         price: "0.00794371",
+            //         total: "7.68882295"
+            //     },
+        }
+        return this.parseTrades (data, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        //         /history
+        //         {
+        //             id: 99601,
+        //             type: "buy",
+        //             time: 1640108002.455,
+        //             amount: "323144.05252658",
+        //             price: "0.008",
+        //             total: "2585.15242021"
+        //         },
+
+        //         /history/result
+        //         {
+        //             tid: 99601,
+        //             type: "buy",
+        //             date: 1640108002,
+        //             amount: "323144.05252658",
+        //             price: "0.008",
+        //             total: "2585.15242021"
+        //         },
+        const symbol = market['symbol'];
+        const id = this.safeString2 (trade, 'id', 'tid');
+        const timestamp = this.safeTimestamp2 (trade, 'time', 'date');
+        const price = this.safeString (trade, 'price');
+        const amount = this.safeString (trade, 'amount');
+        const side = this.safeString (trade, 'type');
+        const takerOrMaker = 'taker'; // as no other info
+        return this.safeTrade ({
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'id': id,
+            'order': undefined,
+            'type': undefined,
+            'takerOrMaker': takerOrMaker,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'cost': undefined,
+            'fee': undefined,
+        }, market);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (market['spot']) {
+            await this.loadMarkets ();
+            const market = this.market (symbol);
+            const request = {
+                'market': market['id'],
+            };
+            if (!(['buy', 'sell'].includes (side))) {
+                throw new InvalidOrder (this.id + ' createOrder side should be either "buy" or "sell"');
+            }
+            if (!(['limit', undefined].includes (type))) {
```

Comment:
Let's remove these checks for valid sides and types, we will unify them in the base classes later: https://github.com/ccxt/ccxt/pull/10114